crates/ethereum/payload/src/lib.rs

```diff
@@ -198,9 +198,15 @@ where
     let mut block_transactions_rlp_length = 0;
 
     let blob_params = chain_spec.blob_params_at_timestamp(attributes.timestamp);
-    let max_blob_count =
+    let protocol_max_blob_count =
         blob_params.as_ref().map(|params| params.max_blob_count).unwrap_or_default();
 
+    // Apply user-configured blob limit (EIP-7872)
+    let max_blob_count = builder_config
+        .max_blobs_per_block
+        .map(|user_limit| std::cmp::min(user_limit as u64, protocol_max_blob_count).max(1))
```

Comment:
is max(1) part of the eip? that this can't be lower than that?
1 doesnt really make much sense really, so imo 0 is also a valid value here